File: backend/src/main/java/com/frosts/testplatform/entity/AlertRule.java
package com.frosts.testplatform.entity;

import com.frosts.testplatform.common.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "sys_alert_rule")
public class AlertRule extends BaseEntity {

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false, length = 30)
    private String ruleType;

    @Column(nullable = false)
    private Boolean enabled = true;

    @Column(name = "metric_name", length = 50)
    private String metricName;

    @Column(name = "condition_type", length = 20)
    private String conditionType;

    private Double threshold;

    @Column(length = 10)
    private String comparator;

    @Column(name = "window_minutes")
    private Integer windowMinutes;

    @Column(name = "min_sample_count")
    private Integer minSampleCount;

    @Column(name = "notify_type", length = 20)
    private String notifyType;

    @Column(length = 20)
    private String priority;

    @Column(name = "cooldown_minutes")
    private Integer cooldownMinutes;

    @Column(columnDefinition = "TEXT")
    private String description;
}

File: frontend/src/api/alertRule.ts
import request from '@/utils/request'

export interface AlertRuleItem {
  id: number
  name: string
  ruleType: string
  enabled: boolean
  metricName: string | null
  conditionType: string | null
  threshold: number | null
  comparator: string | null
  windowMinutes: number | null
  minSampleCount: number | null
  notifyType: string | null
  priority: string | null
  cooldownMinutes: number | null
  description: string | null
  createdAt: string
  updatedAt: string
}

export interface AlertRulePreview {
  currentValue: number | null
  threshold: number
  wouldTrigger: boolean
  sampleCount: number
  message: string
}

export interface CreateAlertRuleParams {
  name: string
  ruleType: string
  metricName?: string
  conditionType?: string
  threshold?: number
  comparator?: string
  windowMinutes?: number
  minSampleCount?: number
  notifyType?: string
  priority?: string
  cooldownMinutes?: number
  description?: string
}

export const getAlertRules = () => {
  return request.get('/alert-rules') as unknown as Promise<{
    code: number
    data: AlertRuleItem[]
  }>
}

export const createAlertRule = (data: CreateAlertRuleParams) => {
  return request.post('/alert-rules', data) as unknown as Promise<{
    code: number
    data: AlertRuleItem
  }>
}

export const updateAlertRule = (id: number, data: Partial<CreateAlertRuleParams>) => {
  return request.put(`/alert-rules/${id}`, data) as unknown as Promise<{
    code: number
    data: AlertRuleItem
  }>
}

export const deleteAlertRule = (id: number) => {
  return request.delete(`/alert-rules/${id}`) as unknown as Promise<{
    code: number
  }>
}

export const toggleAlertRule = (id: number) => {
  return request.patch(`/alert-rules/${id}/toggle`) as unknown as Promise<{
    code: number
    data: AlertRuleItem
  }>
}

export const previewAlertRule = (data: CreateAlertRuleParams) => {
  return request.post('/alert-rules/preview', data) as unknown as Promise<{
    code: number
    data: AlertRulePreview
  }>
}
